src/app/archive_hardfork_toolbox/sql.ml

```diff
@@ -1,8 +1,54 @@
+open Async
 open Core
 open Caqti_request.Infix
 
 module type CONNECTION = Mina_caqti.CONNECTION
 
+module Protocol_version = struct
+  type t = { transaction : int; network : int; patch : int }
+
+  let of_string : string -> t = function
+    | version_str -> (
+        String.split version_str ~on:'.'
```

Comment:
suggest: use scanf to parse protocol version, it'll be simpler and less error prune.